ChromeExtension/js/background.js

```diff
@@ -42,32 +45,27 @@ function createGeniusCurrentLink(e) {
     }
 }
 
-
-
 function createGeniusLink(url) {
-    var req = new XMLHttpRequest();
-    var newLink;
-    req.open('POST', "https://api.geni.us/v2/shorturl?apiKey=" + localStorage["apiKey"] + "&apiSecret=" + localStorage["apiSecret"] + "&format=json&GroupId=" + localStorage["defaultGroupId"] + "&bulkMode=0&domain=geni.us&Url=" + url);
-    req.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
-    req.onload = function () {
-        if (req.status == 200 && req.readyState === 4) {
-            newLink = JSON.parse(req.response)['NewLink'];
+
+ // no entiendo porque no coge los parametros que le envio en el segundo paramentro(spot) de la funcion
+    var client = new GeniusLinkServiceClientPost("https://api.geni.us/v2", localStorage["apiKey"], localStorage["apiSecret"]);
+    client.postToService("shorturl?apiKey=" + localStorage["apiKey"] + "&apiSecret=" + localStorage["apiSecret"] + "&Url=" + url + "&GroupId=" + localStorage["defaultGroupId"] + "&bulkMode=0", 

```

Comment:
We should be passing the parameters in the body no the url and the apiKey and The apiSecret should be added in the internal method.